### ISSUE TITLE
Unhandled rejection Error: unable to verify the first certificate

### ISSUE THREAD

**Issue by MarkHerhold** [coolaj86/nodejs-ssl-example#2]
Hi,
I am mindlessly following the the tutorial and upon starting the server I am greeted with errors.

```
mark@mark-W530:~/nodejs-ssl-example$ node ./serve.js 8043 4080

Redirecting all http traffic to https

Unhandled rejection Error: unable to verify the first certificate
    at Error (native)
    at TLSSocket.<anonymous> (_tls_wrap.js:935:38)
    at emitNone (events.js:67:13)
    at TLSSocket.emit (events.js:166:7)
    at TLSSocket._finishInit (_tls_wrap.js:516:8)
```

My node version (it's io.js):

```
mark@mark-W530:~/nodejs-ssl-example$ node -v
v2.3.4
```

**Comment by coolaj86** [comment]
My best guess is that the certificate that I created the demo with has expired. Looking into it...

In the meantime, this demo works:
- https://github.com/Daplie/localhost.daplie.com-certificates
- https://github.com/Daplie/localhost.daplie.com-server

**Comment by coolaj86** [comment]
~~~Ah, it's because I used the default signature algorithm, which was `sha1WithRSAEncryption` at the time. Since the time I created this demo, `sha1` has been found unsafe and `sha256` is required.~~~

``` bash
openssl x509 -text -noout -in ./certs/server/my-server.crt.pem

openssl x509 -text -noout -in ./certs/ca/my-root-ca.crt.pem
```

I'll just have to regenerate the certificates real quick...

**Comment by MarkHerhold** [comment]
Thanks for the quick reply!

**Comment by coolaj86** [comment]
So I tested that out and it still gave the error.

It turns out that it doesn't like the certificate being served from coolaj86.com for some reason (when it was doing the external ip check). I think there was a glitch in one of the root certificate lists in one of the openssl versions that left out the root signer or something. Chrome had an issue with it a few days ago, but only for a few hours and then it worked again.

Anyway, that's fixed.

**Comment by MarkHerhold** [comment]
Many thanks! I'll give it another go and close the issue when done.

**Comment by coolaj86** [comment]
I ended up learning that the intermediate certificates _do not_ go into the `ca` property, they're supposed to be bundled in with `server.crt` aka `fullchain.pem`. It didn't necessarily affect this repo, it affected the server that was serving the ip check service, but I ended up going back and fixing a number of repos where I had the example incorrect.

`cat cert.pem chain.pem > fullchain.pem`

It's important that the cert comes first and then the intermediates of the chain, but not the root. Some server software requires the keypair to be included as well.

`cat cert.pem chain.pem privkey.pem > combined.pem`
